core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -146,18 +146,17 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
     val topicAndPartition = TopicAndPartition(topic, partition)
     val currState = partitionState.getOrElseUpdate(topicAndPartition, NonExistentPartition)
     try {
+      assertValidTransition(topicAndPartition, targetState)
       targetState match {
         case NewPartition =>
           // pre: partition did not exist before this
```

Comment:
I'm fine with either option. Which do you prefer? By the way, sometimes there are `post` comments as well.